src/utils/Api.js

```diff
@@ -0,0 +1,130 @@
+export default class Api {
+  #url;
+  #headers;
+
+  constructor({ baseUrl, headers }) {
+    this.#url = baseUrl;
+    this.#headers = headers;
+  }
+
+  getAPIInfo() {
+    return Promise.all([this.getUserProfile(), this.getCardsInfo()]);
+  }
+
+  async getCardsInfo() {
+    const res = await fetch(`${this.#url}/cards`, {
+      headers: this.#headers,
+    });
+    if (res.ok) {
+      return res.json();
+    }
+    return await Promise.reject(`Error: ${res.status}`);
```

Comment:
```js
.then(res => {
                if (res.ok) {
                    return res.json();
                }
                return Promise.reject(`Error ${res.status}`);
            })

```

The code of `ok` checking is repeated in **every** request of the project. To avoid it you need to make a special method `_checkResponse` that will check any response from the server if it’s ok or not.


```js
_checkResponse(res) {
        // here is the code of the checking
    }

```

And now you can replace redundant repeating with the only line:


```js
 .then(this._checkResponse)

```

Please, pay attention that you need to pass only the **reference** for the method rather than the call. 

**COULD BE IMPROVED**

You can make a special method (function ) for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this._checkResponse)
  }

```

And now you can safely replace `fetch` with `this._request` and remove repeated checking. And you don’t need to change anything else.

Reply:
Created new function in api.js to handle the heavy lifting, and the called functions call the main function